compiler/util/misc.cpp

```diff
@@ -73,8 +73,14 @@ bool requireWideReferences() {
   return !fLocal || forceWidePtrs();
 }
 
+//
+// If the --no-local flag is used or the locale model is not 'flat'
+// (i.e., has sub-locales that on-clauses might target), we should
+// not drop on-clauses on the floor.  I.e., we should require them
+// to be "outlined."
+//
 bool requireOutlinedOn() {
-  return !fLocal || forceWidePtrs();
+  return !fLocal || strcmp(CHPL_LOCALE_MODEL, "flat");
```

Comment:
Personally I would prefer strcmp(CHPL_LOCALE_MODEL, "flat")==0 because strcmp() really returns an int rather than a bool and the common construct you've coded is always hard for me to read properly because strcmp(a,b) is true when a!=b.  It would be a lot easier to read if the name of the function was strneq() but alas it's not.  Arguably this is C's flaw, not yours, and I know the construct is common, so if you want to leave it that's okay, too.